client/extensions/woocommerce/state/sites/settings/stripe-connect-account/actions.js

```diff
@@ -220,3 +224,124 @@ function deauthorizeFailure( siteId, action, errorMessage ) {
 		siteId,
 	};
 }
+
+/**
+ * Action Creator: Get the initial OAuth URL for connecting a Stripe Account.
+ *
+ * @param {Number} siteId The id of the site for which to create an account.
+ * @param {String} returnUrl The URL for Stripe to return the user to (to complete the setup)
+ * @param {String} [successAction=undefined] Optional action object to be dispatched upon success.
+ * @param {String} [failureAction=undefined] Optional action object to be dispatched upon error.
+ * @return {Object} Action object
+ */
+export const oauthInit = ( siteId, returnUrl, successAction = null, failureAction = null ) => (
+	dispatch,
+	getState
+) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	const initAction = {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_OAUTH_INIT,
+		returnUrl,
+		siteId,
+	};
+
+	dispatch( initAction );
+
+	return request( siteId )
+		.post( 'connect/stripe/oauth/init', { returnUrl }, 'wc/v1' )
+		.then( data => {
+			dispatch( oauthInitSuccess( siteId, initAction, data ) );
+			if ( successAction ) {
+				dispatch( successAction( data ) );
+			}
+		} )
+		.catch( error => {
+			dispatch( oauthInitFailure( siteId, initAction, error ) );
+			if ( failureAction ) {
+				dispatch( failureAction( error ) );
+			}
+		} );
+};
+
+function oauthInitSuccess( siteId, { email }, { oauthUrl } ) {
+	return {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_OAUTH_INIT_COMPLETE,
+		oauthUrl,
+		siteId,
+	};
+}
+
+function oauthInitFailure( siteId, action, { message } ) {
+	return {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_OAUTH_INIT_COMPLETE,
+		error: message,
+		siteId,
+	};
+}
+
+/**
+ * Action Creator: Complete the OAuth flow and connect the Stripe Account.
+ *
+ * @param {Number} siteId The id of the site for which to create an account.
+ * @param {String} stripeCode The hexadecimal code provided by Stripe (GET param) when the user is returned to calypso
+ * @param {String} stripeState The hexadecimal code provided by Stripe (GET param) when the user is returned to calypso
```

Comment:
Are the descriptions of `stripeCode` and `stripeState` supposed to be identical here?

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/19300/commits/77666c56bca1bb9e208934bea5f4769733bc2a8b